libs/designer-ui/src/lib/arrayeditor/util/serializecollapsedarray.ts

```diff
@@ -71,11 +71,11 @@ export const parseSimpleItems = (
   // Beautify ValueSegment
   try {
     const nodeMap = new Map<string, ValueSegment>();
-    const stringValueUncasted = JSON.stringify(JSON.parse(convertSegmentsToString(uncastedArraySegments, nodeMap)), undefined, 4);
     const stringValueCasted = JSON.stringify(JSON.parse(convertSegmentsToString(castedArraySegments, nodeMap)), undefined, 4);
+    const stringValueUncasted = JSON.stringify(JSON.parse(convertSegmentsToString(uncastedArraySegments, nodeMap)), undefined, 4);
```

Comment:
Nothing to block this PR but we should create a helper function to stringify the json properly and use it in all the places to make it easier to maintain and add a particular note explaining why we need to do this.

basically, something like - 

`const stringifyJson: (json: string) => JSON.stringify(JSON.parse(json), undefined, 4);`